prov/efa/src/rdm/rxr_msg.c

```diff
@@ -580,7 +580,17 @@ int rxr_msg_handle_unexp_match(struct rxr_ep *ep,
 		rx_entry->ignore = ~0;
 	}
 
-	return rxr_pkt_proc_matched_rtm(ep, rx_entry, pkt_entry);
+	srx = rx_entry->peer_rx_entry.srx;
+
+	// TODO - populate rx_entry->fi_peer_rx_entry with info required by the SHM provider
```

Comment:
I think shm provider does similar thing, but it uses peer_context: https://github.com/ofiwg/libfabric/blob/main/prov/shm/src/smr_progress.c#L769-L784. So this shouldn't be an issue. EFA provider should not care how shm use peer_rx_entry

Reply:
The `owner_context` seems to be filled as the owner entry of multi-recv buffer: https://github.com/ofiwg/libfabric/blob/main/prov/shm/src/smr_ep.c#L1411-L1426. Though there is no definition of this field in the man page. Maybe open an issue to ask Intel for clarification?

Reply:
I started a discussion https://github.com/ofiwg/libfabric/discussions/8646 and emailed Alexia

Reply:
Cool, it seems using `owner_context` is good here. Thanks!

Reply:
Back to your `TODO:`, now I think you should populate the info in `rx_entry` to `rx_entry->peer_rx_entry`.  I think they should be implemented somewhere here: https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rdm/rxr_msg.c#L860-L871 before calling `rxr_msg_handle_unexp_match`

Reply:
> you should populate the info in `rx_entry` to `rx_entry->peer_rx_entry` in unexp path

Only SHM provider needs this info. EFA provider will just use `rx_entry`. So I didn't implement that now and added the `TODO` instead. I can add the info when implementing the SHM provider?

Reply:
I see, sure. It can be in a separate commit when integrating with shm.